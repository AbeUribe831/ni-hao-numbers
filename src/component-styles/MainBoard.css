.welcome-text {
    font-size: 3em;
    text-align: center;
}

.main-board {
  /*background-color: rgb(82, 80, 80);*/ 
  background-color: rgb(19, 18, 18);
  display: flex;
  overflow-y: scroll;
  justify-content: center;
  padding-top: 0.25em;
  height: 100%;
  width: 100%;
}
.basic-button {
    padding: 0em;
    width: 4em;
    height: 2em;
    background: #cd071e;
    border: none;
    border-radius: 5%;
    color: white;
    cursor: pointer;
}

/* use margin to move background and mic button*/
.mic-background {
    width:60px;
    height:60px;
    background-color:red;
    border-radius:2em;
}
/* Mic button */
.gg-mic {
  box-sizing: border-box;
  position: absolute;
  display: block;
  transform: scale(var(--ggs,1));
  width: 20px;
  height: 16px;
  border-bottom-left-radius: 120px;
  border-bottom-right-radius: 120px;
  border: 2px solid;
  border-top: 0px;
  margin-top: 23px;
  margin-left: 20px;
}
  
.gg-mic::after,
.gg-mic::before {
  content: "";
  display: block;
  box-sizing: border-box;
  position: absolute;
}
  
.gg-mic::after {
  border: 2px solid;
  width: 12px;
  height: 29px;
  left: 2px;
  top: -16px;
  border-radius: 5px;
}
/*Bottom of mic */
.gg-mic::before {
  width: 11px;
  height: 6px;
  top: 16px;
  left: 2px;
  border-right: 4px solid transparent;
  box-shadow:
  0 2px 0,
  inset -2px 0 0
} 
/* play button width: 22, height: 22 */
.gg-play-button {
  border: none;
  box-sizing: border-box;
  position: relative;
  display: block;
  transform: scale(var(--ggs,1));
  width: 40px;
  height: 29px;
  color: white;
  background-color: #f00f1e;
  border-radius: 0.6em;
}
  
.gg-play-button::before {
  cursor: pointer;
  content: "";
  display: block;
  box-sizing: border-box;
  position: absolute;
  width: 0;
  height: 20px;
  border-top: 10px solid transparent;
  border-bottom: 9px solid transparent;
  border-left: 12px solid;
  top: 5px;
  left: 16px
} 