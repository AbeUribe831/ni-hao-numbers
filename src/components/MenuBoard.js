import React, { Component } from 'react'
import '../component-styles/MenuBoard.css'
import '../component-styles/MainBoard.css'
// TODO:: add ID to each button
class MoreOptions extends Component {
    constructor(props){
        super(props);
        this.rChar = 'readCharacter';
        this.rNum = 'readNumber';
        this.listen = 'listen';
    }
    
    render(){
        
        const which_button = this.props.isMobile === false ? 'standard-button' : 'mobile-standard-button';
        return(
            <div className='flex-column'>
                    <p className={this.props.isMobile === false ? 'desktop-sub-titles': ''} 
                        style={{
                            margin: '8px 0px 0px 0px',
                            alignSelf:'center', 
                            color: 'rgb(235, 200, 5)'
                        }}>Questions</p>
                    <div className='flex-row margin-top'>
                        <button className={which_button} 
                            style={{
                                transitionDuration: '0.1s',
                                backgroundColor: this.props.questions.readCharacter ? '#cd071e' : 'white',
                                color: this.props.questions.readCharacter ? 'white' : '#cd071e'}} 
                            onClick={() => this.props.updateQuestions(this.rChar)}
                        >
                            read character
                        </button>
                        <div style={{marginLeft: '0.5em'}}></div>
                        <button className={which_button}
                            style={{
                                transitionDuration: '0.1s',
                                backgroundColor: this.props.questions.readNumber ? '#cd071e' : 'white',
                                color: this.props.questions.readNumber ? 'white' : '#cd071e'}} 
                            onClick={() => this.props.updateQuestions(this.rNum)}
                        >
                            read number
                        </button>
                        <div style={{marginLeft: '0.5em'}}></div>
                        <button className={which_button}
                            style={{
                                transitionDuration: '0.1s',
                                backgroundColor: this.props.questions.listen ? '#cd071e' : 'white',
                                color: this.props.questions.listen ? 'white' : '#cd071e'}} 
                            onClick={() => this.props.updateQuestions(this.listen)}
                        >
                            listen
                        </button>
                    </div>
                <div className={'margin-top'} style={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
                    <button onClick={this.props.onClickMoreOptions} className='underline-button' style={{textAlign: 'center'}}>hide</button>
                </div>
            </div>
        );
    }
}

function NumberOptions(props) {
    return (
        <div className='flex-column' >
            <div className='flex-row margin-top'>
                <div className='flex-column'>            
                    <p className={props.isMobile === true ? 'sub-titles' : 'desktop-sub-titles'}>From</p>
                    <input className={props.isMobile === true ? '' : 'desktop-input'}
                        name={'minBound'} 
                        value={props.minBound} 
                        onChange={props.updateMinBound} 
                        onKeyDown={props.confirmMinIsNumber}
                    />
                </div>
                <div style={{marginLeft: '1em'}}></div>
                <div className='flex-column'>
                    <p className={props.isMobile === true ? 'sub-titles' : 'desktop-sub-titles'} >To</p>
                    <input className={props.isMobile === true ? '' : 'desktop-input'}
                        name={'maxBound'}
                        value={props.maxBound}    
                        onChange={props.updateMaxBound}
                        onKeyDown={props.confirmMaxIsNumber}
                    />
                </div>
                <div style={{marginLeft: '1em'}}></div>
                <div className='flex-column'>
                    <p className={props.isMobile === true ? 'sub-titles' : 'desktop-sub-titles'}>How Many</p>
                    <input className={props.isMobile === true ? '' : 'desktop-input'}
                        name={'howMany'}
                        value={props.howMany}
                        onChange={props.updateHowMany}
                        onKeyDown={props.confirmHowMany}
                    />
                </div>
            </div>
            <div className='flex-row margin-top'>
                <div className='flex-column'>
                    <p className={props.isMobile === true ? 'sub-titles' : 'desktop-sub-titles'}>Decimal</p>
                    <select className={props.isMobile === true ? '' : 'desktop-input'}
                        name="decimal" value={props.decimalPlacement} onChange={props.updateDecimalPlacement} id="decimal">
                        <option value="0">0</option>
                        <option value="1">1</option>
                        <option value="2">2</option>
                    </select>
                </div>
                <div style={{marginLeft: '1em'}}></div>
                <div className='flex-column'>
                    <p className={props.isMobile === true ? 'sub-titles' : 'desktop-sub-titles'}>Characters</p>
                    <select className={props.isMobile === true ? '' : 'desktop-input'}
                        name="chinese-character-type" value={props.chnCharType} onChange={props.updateChnCharType} id="chinese-character-type" >
                        <option value="sc">简体</option>
                        <option value="tc">繁體</option>
                    </select>
                </div>
            </div>
        </div>
    );
}

function Options (props) {
   const which_button = props.isMobile === false ? 'standard-button' : 'mobile-standard-button';
   if (props.moreOptions === false) {
       return (
           <div className='flex-column' style={(props.isMobile === false) ? {width: '100%'}: {}}>
                <NumberOptions
                    isMobile={props.isMobile}
                    minBound={props.minBound}
                    updateMinBound={props.updateMinBound}
                    confirmMinIsNumber={props.confirmMinIsNumber}
                    maxBound={props.maxBound}
                    decimalPlacement={props.decimalPlacement}
                    chnCharType={props.chnCharType}
                    updateMaxBound={props.updateMaxBound}
                    updateDecimalPlacement={props.updateDecimalPlacement}
                    confirmMaxIsNumber={props.confirmMaxIsNumber}
                    howMany={props.howMany}
                    updateHowMany={props.updateHowMany}
                    updateChnCharType={props.updateChnCharType}
                    confirmHowMany={props.confirmHowMany}
                />
                <div className='margin-top' style={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
                    <button onClick={props.onClickMoreOptions} className='underline-button'>
                        more options
                    </button>
                </div>
                <div style={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
                    <button className={which_button}
                        style={{color: 'white', backgroundColor: '#cd071e', type: 'button', cursor: 'pointer'}}
                        onClick={props.onClickStart}
                    >
                        start
                    </button>
                </div>
           </div>
       )
   } 
   return (
            <div className="flex-column" style={(props.isMobile === false) ? {width: '100%'}: {}}>
                 <NumberOptions
                    isMobile={props.isMobile}
                    minBound={props.minBound}
                    updateMinBound={props.updateMinBound}
                    confirmMinIsNumber={props.confirmMinIsNumber}
                    maxBound={props.maxBound}
                    updateDecimalPlacement={props.updateDecimalPlacement}
                    updateMaxBound={props.updateMaxBound}
                    confirmMaxIsNumber={props.confirmMaxIsNumber}
                    howMany={props.howMany}
                    updateHowMany={props.updateHowMany}
                    confirmHowMany={props.confirmHowMany}
                />
                <div style={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
                    <MoreOptions 
                        isMobile={props.isMobile}
                        questions={props.questions} 
                        answers={props.answers} 
                        started={props.started}
                        onClickMoreOptions={props.onClickMoreOptions} 
                        updateQuestions={props.updateQuestions}
                        updateAnswers={props.updateAnswers}
                        resetQAndA={props.resetQAndA}
                    />
                </div>
                <div style={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
                    <button className={which_button}
                        style={{color: 'white', backgroundColor: '#cd071e', type: 'button', cursor: 'pointer'}}
                        onClick={props.onClickStart}
                    >
                        start
                    </button>
                </div>
           </div>
   )
}
// props: {questions, answers}
export default class MenuBoard extends Component {
    constructor(props) {
        super(props);
        this.state = {
            moreOptions: false,
        }
        this.onClickMoreOptions = this.onClickMoreOptions.bind(this);
    }
    onClickMoreOptions() {
        this.setState((prevState) => {
            return ({
                    moreOptions: !prevState.moreOptions
            });
        });
        this.props.resetQAndA();
    }
    // in from add : <form onClick={this.method}
    render() {
        if(this.props.isMobile === true) {
            return (
            <div >
                <div className='welcome-text'>
                    准备好了吗？
                </div>
                <div>
                    <Options 
                        isMobile={this.props.isMobile}
                        minBound={this.props.minBound}
                        maxBound={this.props.maxBound}
                        howMany={this.props.howMany}
                        questions={this.props.questions} 
                        answers={this.props.answers}
                        started={this.props.started}
                        decimalPlacement={this.props.decimalPlacement}
                        chnCharType={this.props.chnCharType}
                        onClickMoreOptions={this.onClickMoreOptions} 
                        moreOptions={this.state.moreOptions} 
                        updateDecimalPlacement={this.props.updateDecimalPlacement}
                        updateQuestions={this.props.updateQuestions}
                        updateAnswers={this.props.updateAnswers}
                        updateMinBound={this.props.updateMinBound}
                        updateMaxBound={this.props.updateMaxBound}
                        updateHowMany={this.props.updateHowMany}
                        updateChnCharType={this.props.updateChnCharType}
                        confirmMinIsNumber={this.props.confirmMinIsNumber}
                        confirmMaxIsNumber={this.props.confirmMaxIsNumber}
                        confirmHowMany={this.props.confirmHowMany}
                        resetQAndA={this.props.resetQAndA}
                        onClickStart={this.props.onClickStart}
                    />
                    </div>
                </div>
            )
        }
        return (
                
            <div style={{width: '100%', height: '100%', display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'center'}}>
                <div className='welcome-text' style={{display:'flex', flex:'1',alignItems:'flex-end', color: 'white', textAlign: 'center', paddingBottom: '0px', fontSize:'4em'}}>
                    准备好了吗？
                </div>
                <div style={{flex:'3'}}>
                    <Options 
                        isMobile={this.props.isMobile}
                        minBound={this.props.minBound}
                        maxBound={this.props.maxBound}
                        howMany={this.props.howMany}
                        questions={this.props.questions} 
                        answers={this.props.answers}
                        started={this.props.started}
                        decimalPlacement={this.props.decimalPlacement}
                        chnCharType={this.props.chnCharType}
                        onClickMoreOptions={this.onClickMoreOptions} 
                        moreOptions={this.state.moreOptions} 
                        updateDecimalPlacement={this.props.updateDecimalPlacement}
                        updateQuestions={this.props.updateQuestions}
                        updateAnswers={this.props.updateAnswers}
                        updateMinBound={this.props.updateMinBound}
                        updateMaxBound={this.props.updateMaxBound}
                        updateHowMany={this.props.updateHowMany}
                        updateChnCharType={this.props.updateChnCharType}
                        confirmMinIsNumber={this.props.confirmMinIsNumber}
                        confirmMaxIsNumber={this.props.confirmMaxIsNumber}
                        confirmHowMany={this.props.confirmHowMany}
                        resetQAndA={this.props.resetQAndA}
                        onClickStart={this.props.onClickStart}
                    /></div>
                </div>
        )
    }
}