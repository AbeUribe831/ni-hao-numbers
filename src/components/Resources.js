import React from 'react'
import '../component-styles/MainBoard.css'
export default function Resources() {
    return (
        <div className='main-board' style={{flex: '2'}}>
        <h3 style={{textAlign : 'center'}}>Resources</h3>
        </div>

    )
}