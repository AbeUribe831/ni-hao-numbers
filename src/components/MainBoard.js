import React, { Component } from 'react'
import '../component-styles/MainBoard.css'
import { MenuBoard } from './MenuBoard'
import StudyBoard from './StudyBoard'

// add a loading state 
export default class MainBoard extends Component {
    constructor(props){
        super(props);
        // the values in questions and answers determine which options are pressed
        this.state = {
            minBound: '0',
            maxBound: '10',
            howMany: '15',
            decimalPlacement: '0',
            chnCharType: 'sc',
            questions: {
                readCharacter: true,
                readNumber: true,
                listen: true
            },
            answers: {speak: false},
            loading: false,
            started: false 
        }
        this.updateQuestions = this.updateQuestions.bind(this);
        this.updateAnswers = this.updateAnswers.bind(this);
        this.resetQAndA = this.resetQAndA.bind(this);
        this.updateDecimalPlacement = this.updateDecimalPlacement.bind(this);
        this.updateMinBound = this.updateMinBound.bind(this);
        this.updateMaxBound = this.updateMaxBound.bind(this);
        this.updateHowMany = this.updateHowMany.bind(this);
        this.updateLoading = this.updateLoading.bind(this);
        this.updateChnCharType = this.updateChnCharType.bind(this);
        this.confirmMinIsNumber = this.confirmMinIsNumber.bind(this);
        this.confirmMaxIsNumber = this.confirmMaxIsNumber.bind(this);
        this.confirmHowMany = this.confirmHowMany.bind(this);
        this.onClickStartedTrue = this.onClickStartedTrue.bind(this);
        this.onClickStartedFalse = this.onClickStartedFalse.bind(this);
    }

    // TODO:: modify to work with largest and smallest number
    confirmMaxIsValid() {
        const minBound = parseFloat(this.state.minBound);
        const maxBound =  parseFloat(this.state.maxBound);
            if (minBound >= maxBound) {
                this.setState((prevState) => ({
                    maxBound: (prevState.minBound + 1).toString()
                }));
            }
    }
    /*
    All these confirm methods are for clicking Tab or Enter
    */
    confirmMinIsNumber(event) {
        if ((event.key === 'Tab' || event.key ==='Enter') && (this.state.minBound === '' || isNaN(this.state.minBound))) {
            this.setState({
                minBound: '0'
            });
        }
    }
    confirmMaxIsNumber(event) {
        if ((event.key === 'Tab' || event.key ==='Enter') && (this.state.maxBound === '' || isNaN(this.state.maxBound) || parseFloat(this.state.minBound) >= parseFloat(this.state.maxBound))) {
            this.setState((prevState) => ({
                maxBound: (parseFloat(prevState.minBound) + 1).toString()
            }));
 
        }
    }
    confirmHowMany(event) {
        if ((event.key === 'Tab' || event.key ==='Enter') && (this.state.howMany === '' || parseInt(this.state.howMany)  < 1)) {
            this.setState({
                howMany: '1'
            });
        }
    }
    /*
    Update methodschecks that the input is a number or empty (empty to allow user to type in a new number)
    */
    updateHowMany(event) {
        let allNumRegex = /^[\d]*$/g;
        if (event.target.value.match(allNumRegex)) {
            this.setState({
                howMany: event.target.value
            }, () => {
                if (parseInt(this.state.howMany) > 50) {
                    this.setState({
                        howMany: '50'
                    });
                }
            });
        }
        else {
            this.setState({
                howMany: '1'
            });
        }
    }
    // TODO:: add a limit to min -9,999,999,999,999.99
    updateMinBound(event)
    {
        let allNumRegex = /^[-]{0,1}[\0\d]*[.]{0,1}[\0\d]*$/g;
        if (event.target.value.match(allNumRegex)) {
            this.setState({
                minBound: event.target.value
            }, () => {
                this.confirmMaxIsValid()
            });
        }
        else {
            this.setState({
                minBound: '0'
            });
        }
    }
    // TODO:: add a limit to max 9,999,999,999,999.99
    updateMaxBound(event)
    {
        // let regex = /^[-]{0,1}[\0\d]*$/g;
        let allNumRegex = /^[-]{0,1}[\0\d]*[.]{0,1}[\0\d]*$/g;
        if (event.target.value.match(allNumRegex)) {
            this.setState({
                maxBound: event.target.value
            });
        }
        else {
            this.setState((prevState) => ({
                maxBound: (parseFloat(prevState.minBound) + 1).toString()
            }));
        }
    }
    // TODO:: determine whether to toggle the loading or set true/false
    updateLoading() {
        this.setState((prevState) =>({
            loading: !prevState.loading
        }));
    }
    resetQAndA() {
        this.setState({
            questions: {
                readCharacter: true,
                readNumber: true,
                listen: true,
            },
            answers: {speak: false}
        });
    }
    updateDecimalPlacement(event) {
        this.setState({
            decimalPlacement: event.target.value 
        });
    }
    updateQuestions(question) {
        if (question === 'readCharacter') {
            this.setState((prevState) =>({
                questions: {
                    readCharacter: !prevState.questions.readCharacter,
                    readNumber: prevState.questions.readNumber,
                    listen: prevState.questions.listen
                }
            }));
        }
        else if (question === 'readNumber') {
            this.setState((prevState) =>({
                questions: {
                    readCharacter: prevState.questions.readCharacter,
                    readNumber: !prevState.questions.readNumber,
                    listen: prevState.questions.listen
                }
            }));
        }
        else if (question === 'listen') {
            this.setState((prevState) =>({
                questions: {
                    readCharacter: prevState.questions.readCharacter,
                    readNumber: prevState.questions.readNumber,
                    listen: !prevState.questions.listen
                }
            }));
        }
    }
    updateAnswers() {
        this.setState((prevState) => ({
            answers: {speak: !prevState.answers.speak}
        }));
    }
    updateChnCharType(event) {
        this.setState({
            chnCharType: event.target.value
        });
    }
    onClickStartedTrue() {
        this.setState({
            started: true
        });
        if(this.props.isMobile) {
            this.props.hideMobileNav(true);
        }
    }
    onClickStartedFalse() {
        this.setState({
            started: false 
        });
        if(this.props.isMobile) {
            this.props.hideMobileNav(false);
        }
    }
    render() {
        const started = this.state.started;
        if(!started) { 
            return (
                <div className='main-board' style={{paddingTop: '3em'}}>
                    <MenuBoard 
                        isMobile={this.props.isMobile}
                        minBound={this.state.minBound}
                        maxBound={this.state.maxBound}
                        howMany={this.state.howMany}
                        questions={this.state.questions} 
                        answers={this.state.answers}
                        started={this.state.started}
                        decimalPlacement={this.state.decimalPlacement}
                        chnCharType={this.state.chnCharType}
                        updateDecimalPlacement={this.updateDecimalPlacement}
                        updateQuestions={this.updateQuestions} 
                        updateAnswers={this.updateAnswers}
                        updateMinBound={this.updateMinBound}
                        updateMaxBound={this.updateMaxBound}
                        updateHowMany={this.updateHowMany}
                        updateChnCharType={this.updateChnCharType}
                        confirmMinIsNumber={this.confirmMinIsNumber}
                        confirmMaxIsNumber={this.confirmMaxIsNumber}
                        confirmHowMany={this.confirmHowMany}
                        resetQAndA={this.resetQAndA}
                        onClickStart={this.onClickStartedTrue}
                    />
                </div>
            )
        }
        return (
            <div className='main-board' style={{paddingTop: '5em'}}>
                <StudyBoard
                    isMobile={this.props.isMobile}
                    minBound={this.state.minBound}
                    maxBound={this.state.maxBound} 
                    howMany={this.state.howMany}
                    questions={this.state.questions}
                    answers={this.state.answers}
                    loading={this.state.loading}
                    chnCharType={this.state.chnCharType}
                    decimalPlacement={this.state.decimalPlacement}
                    onClickExit={this.onClickStartedFalse}
                    updateLoading={this.updateLoading}
                    resetQAndA={this.resetQAndA}
                />
            </div>
        )
    }
}