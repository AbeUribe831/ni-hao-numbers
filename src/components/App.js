import React, {Component} from 'react';
import '../component-styles/App.css';
import MainBoard from './MainBoard'
import Resources from './Resources'
import {
    BrowserRouter,
    Link,
    Route
} from 'react-router-dom'

// App will manage the state to pass between routes, including login

// Have nav component with signin button for signin component and home page
// main page is main board 
// signin is signup component
// TODO:: add a css class to add a container for the main board / signin
// change Eng button to select div

// will treat (0, 1224px] as mobile / tablet
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      is_mobile: window.matchMedia("(max-width: 800px)").matches,
    }
    this.highlightClickedPage = this.highlightClickedPage.bind(this); 
  }
  componentDidMount() {
    const handler = e => this.setState({is_mobile: e.matches});
    window.matchMedia("(max-width: 800px)").addEventListener('change', handler);
  }
  // changle color to the current page
  highlightClickedPage(table_id) {
    var table_row = document.getElementById('mobile-nav-bar-table').rows[0];
    console.log(table_id)
    console.log(table_row);
    for (var i = 0; i < table_row.cells.length; i++) {
      if(table_id === table_row.cells[i].id) {
        table_row.cells[i].lastChild.style.color = '#cd071e';
      }
      else {
        table_row.cells[i].lastChild.style.color = 'white';
      }
    }
  }
  // add flex of main body in the main board
  render() {
  return (
    <div style={{height: '100%'}}>
      {!this.state.is_mobile && (
        <p>not mobile</p>
      )}
      {this.state.is_mobile && (
        <div style={{display: 'flex', flexDirection: 'column', height: '100%'}}>
          <div className='header-bar' style={{flex:'0 1 3em'}}>
            <p style={{fontFamily:'KaiTi, sans-serif', textAlign: 'center', color: 'white'}}>你 好 numbers</p>
          </div>
            <BrowserRouter>
              <Route exact path='/' component={MainBoard}/>
              <Route path='/resources' component={Resources}/> 
             <nav>
                <table id='mobile-nav-bar-table' style={{flex:'1', paddingBottom: '0.5em'}}className='mobile-nav-bar'>
                  <tbody>
                    <tr>
                      <th id='home-th' ><Link onClick={() => this.highlightClickedPage('home-th')} style={{color: '#cd071e'}} to='/'>Home</Link></th>
                      <th id='resources-th'><Link onClick={() => this.highlightClickedPage('resources-th')} style={{color: 'white'}} to='/resources'>Resources</Link></th>
                    </tr>
                  </tbody>
                </table>
              </nav>
            </BrowserRouter>
        </div>
      )}
   </div>
  );
  }
}
/*
function App() {
  return (
    <div>
     <ul className="nav-bar">
            <BrowserRouter>
                <nav>
                    <li className="nav-bar-item black-hover"><Link className="nav-bar-link" to='/'>Ni Hao Hao</Link></li>
                    <li className="float-right end"><Link className="nav-bar-link"to='/signin'>Resources</Link></li>
                    <li className="float-right black-hover"><Link className="nav-bar-link"to='/eng'>Eng</Link></li>
                </nav>

                <Route exact path='/' component={MainBoard}/>
                <Route path='/signin' component={Resources}/>
            </BrowserRouter>
        </ul> 
    </div>
  );
}
*/
export default App;
