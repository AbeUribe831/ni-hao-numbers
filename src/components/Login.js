import '../App.css'
export default function Login() {
    return (
        <div className='main-board'>
        <h3 style={{textAlign : 'center'}}>login</h3>
        </div>

    )
}