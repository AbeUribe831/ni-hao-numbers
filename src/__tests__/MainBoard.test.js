/**
 * @jest-environment jsdom
 */
import React from 'react';
import { MainBoard } from '../components/MainBoard';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

let main_board_props;

beforeEach(() => {
    main_board_props = {
        isMobile: false,
        hideMobileNav: jest.fn()
    };
});

afterEach(() => {
    jest.clearAllMocks();
});

describe('rendering components', () => {
    test('rendering MainBoard', () => {
        Enzyme.shallow(
            <MainBoard
                {...main_board_props}
            />
        );
    });
});

describe('passing props', () => {
    test('MainBoard passing props', () => {
        const wrapper = Enzyme.mount(
            <MainBoard
                {...main_board_props}
            />
        );
        expect(wrapper.props().isMobile).toBe(false);
        expect(typeof wrapper.props().hideMobileNav === typeof jest.fn()).toBe(true);
    });
});
// TODO:: verify all logic for inputs (Min bounds, max bounds, etc.) 
// TODO:: show when started does change from MenuBoard to MainBoard component
describe('logic', () => {})

describe('snapshots', () => {
    test('renders MainBoard containing MenuBoard component when started is false', () => {
        const wrapper = Enzyme.shallow(
            <MainBoard
                {...main_board_props}
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('renders MainBoard containing StudyBoard when started is true', () => {
        const wrapper = Enzyme.shallow(
            <MainBoard
                {...main_board_props}
            />
        );
        wrapper.setState({
            started: true
        })
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})