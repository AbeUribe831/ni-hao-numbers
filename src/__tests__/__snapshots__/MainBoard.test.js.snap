// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots renders MainBoard containing MenuBoard component when started is false 1`] = `
<div
  className="main-board"
  style={
    Object {
      "paddingTop": "3em",
    }
  }
>
  <MenuBoard
    answers={
      Object {
        "speak": false,
      }
    }
    chnCharType="sc"
    confirmHowMany={[Function]}
    confirmMaxIsNumber={[Function]}
    confirmMinIsNumber={[Function]}
    decimalPlacement="0"
    howMany="15"
    isMobile={false}
    maxBound="10"
    minBound="0"
    onClickStart={[Function]}
    questions={
      Object {
        "listen": true,
        "readCharacter": true,
        "readNumber": true,
      }
    }
    resetQAndA={[Function]}
    started={false}
    updateAnswers={[Function]}
    updateChnCharType={[Function]}
    updateDecimalPlacement={[Function]}
    updateHowMany={[Function]}
    updateHowManyBlur={[Function]}
    updateMaxBound={[Function]}
    updateMaxBoundBlur={[Function]}
    updateMinBound={[Function]}
    updateMinBoundBlur={[Function]}
    updateQuestions={[Function]}
  />
</div>
`;

exports[`snapshots renders MainBoard containing StudyBoard when started is true 1`] = `
<div
  className="main-board"
  style={
    Object {
      "paddingTop": "5em",
    }
  }
>
  <StudyBoard
    answers={
      Object {
        "speak": false,
      }
    }
    chnCharType="sc"
    decimalPlacement="0"
    howMany="15"
    isMobile={false}
    loading={false}
    maxBound="10"
    minBound="0"
    onClickExit={[Function]}
    questions={
      Object {
        "listen": true,
        "readCharacter": true,
        "readNumber": true,
      }
    }
    resetQAndA={[Function]}
    updateLoading={[Function]}
  />
</div>
`;
